spec/02-integration/02-cmd/02-start_stop_spec.lua

```diff
@@ -353,6 +354,8 @@ describe("kong start/stop #" .. strategy, function()
       assert.res_status(404, res) -- no Route configured
       assert(helpers.stop_kong(helpers.test_conf.prefix))
 
+      ngx.sleep(1)  -- XXX: why?
```

Comment:
`helpers.stop_kong stops` Kong by exec `kill --signal SIGTERM kong_pid`. This signal does not seem to guarantee that the process will be terminated immediately.

Please see [Kong/lua-resty-events/lualib/resty/events/queue.lua#L39](https://github.com/Kong/lua-resty-events/blob/439d5945e21b95ec129574d261546109ec3dfcac/lualib/resty/events/queue.lua#L39)
I guess it may be caused by this `semaphore:wait(1)`.

It's better to test it.

Reply:
Does it seem that there is no other better solution?

Reply:
https://github.com/Kong/kong/blob/0973036503941c52c56f8a1be74b686a62a40cdd/spec/02-integration/02-cmd/08-quit_spec.lua#L19-L35

Maybe we can replace `helpers.stop_kong` with graceful shutdown.

Reply:
Good idea, Let's try it.